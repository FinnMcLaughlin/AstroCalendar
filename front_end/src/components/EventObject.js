import React, { Component } from 'react'
import PropTypes from 'prop-types';

export class EventObject extends Component {

    getStyle = () => {
        return({
            backgroundColor: this.props.event_info.reminder_set ? '#db1616' : '#f4f4f4'
        })
    }

    render() {
        const { event_name, event_date, id } = this.props.event_info;
        
        return (
            <div data-testid="event-obj-div" style= { this.getStyle() }>
                <h2>{event_name}</h2>
                <h3>{event_date}</h3>
                <button onClick={this.props.setReminder.bind(this, id)}>Create Reminder</button>
            </div>
        )
    }
}


// PropTypes
EventObject.propTypes = {
    event_info: PropTypes.object.isRequired,
    setReminder: PropTypes.func.isRequired
}

export default EventObject
