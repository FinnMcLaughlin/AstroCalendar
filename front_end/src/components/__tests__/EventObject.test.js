import '@testing-library/jest-dom';
import { render, cleanup, fireEvent} from '@testing-library/react'
import EventObject from '../EventObject'

afterEach(() => {
    cleanup()
})

// Reminder Button Label
const reminder_button_label = "Create Reminder"

// Mock event object to be passed to /EventObject
const test_event_obj = {
    id: 1,
    event_name: "St. Patricks Day",
    event_date: "17/03/2021",
    reminder_set: true
}

// Event Reminder array to test the correct ID was returned from /EventObject
const test_event_reminders = []

// Mock function to populate test_event_reminder array with the event object id when button clicked
function setReminder(event_id){
    console.log("Reminder set for event: ", event_id)
    test_event_reminders.push(event_id)
}

test("renders event object", () => {

    const { getByTestId } = render(<EventObject event_info={test_event_obj} setReminder={setReminder}/>)

    const event_div = getByTestId("event-obj-div")
    
    expect(event_div).toMatchSnapshot()
})

test("button sets reminder for correct event", () => {

    const { getByText } = render(<EventObject event_info={test_event_obj} setReminder={setReminder}/>)

    expect(test_event_reminders.length).toEqual(0)

    fireEvent.click(getByText(reminder_button_label))

    expect(test_event_reminders.length).toEqual(1)
    expect(test_event_reminders).toContain(1)
})