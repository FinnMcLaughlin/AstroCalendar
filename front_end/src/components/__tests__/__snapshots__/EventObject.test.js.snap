// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders event object 1`] = `
<div
  data-testid="event-obj-div"
  style="background-color: rgb(219, 22, 22);"
>
  <h2>
    St. Patricks Day
  </h2>
  <h3>
    17/03/2021
  </h3>
  <button>
    Create Reminder
  </button>
</div>
`;
